src/Dialog/Dialog.tsx

```diff
@@ -96,12 +101,14 @@ export interface DialogProps extends SxProp {
   footerButtons?: DialogButtonProps[]
 
   /**
-   * This method is invoked when a gesture to close the dialog is used (either
-   * an Escape key press or clicking the "X" in the top-right corner). The
-   * gesture argument indicates the gesture that was used to close the dialog
-   * (either 'close-button' or 'escape').
+   * This method is invoked when a gesture to close the dialog is used
+   * (either an Escape key press, clicking/tapping on the backdrop,
+   * clicking/tapping the 'X' in the top-right corner or dragging away a
+   * bottom sheet). The gesture argument indicates the gesture that was
+   * used to close the dialog.
    */
-  onClose: (gesture: 'close-button' | 'escape') => void
+
+  onClose: (gesture: 'close-button' | 'escape' | 'drag' | 'overlay') => void
```

Comment:
Some concern as I posted in `ConfirmationDialog`:

> It could be confusing if somebody tries to pass 'drag' or 'overlay' on something that doesn't use those gestures. I'm not sure of a better way to handle those, but maybe we can explicitly mention which variants accept which gestures in the JSDoc comment and prop docs.

Reply:
I believe it's clearer in this context because the dialogue supports it with the bottom sheet variant. But I'm curious if those gestures are currently in used. Why would you need to know how the user closed the dialogue? What do you think...

Reply:
We should avoid extending these. The gestures aren't checked in consuming code, and in the future the `onClose()` call will be called whenever the dialog is closed and will not pass a gesture back. See https://github.com/primer/react/pull/3840/files#diff-7c9e4f2540e063c93f543e6eaa44cac7dc3b0e55a8fe81030c9e571f72950a3eR97-R102

Reply:
@keithamus Makes a lot of sense. I've removed it and always return `close-button` for now.